src/query/sql/src/planner/optimizer/rule/rewrite/rule_push_down_limit_aggregate.rs

```diff
@@ -45,45 +48,182 @@ impl RulePushDownLimitAggregate {
     pub fn new() -> Self {
         Self {
             id: RuleID::PushDownLimitAggregate,
```

Comment:
It's better to give the rule a new name.

Reply:
How about `RulePushDownRankLimitAggregate` ?

Reply:
make sense